hatice-ecevit-2016400138/assignment4/countIslands.java

```diff
@@ -0,0 +1,32 @@
+
+public class countIslands {
+    /**
+     *
+     * @param map is the map of tiles given.
+     * @param row number of rows.
+     * @param column number of columns.
+     * @return -1 if the map is null, number of islands otherwise.
+     */
+    public static int findNumOfIslands (boolean[][] map, int row, int column) {
+        if (map == null)
+            return -1;
+        int numOfIslands = 0;
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < column; j++) {
+                if (map[i][j]) {
+                    if (i > 0 && j > 0 && map[i-1][j] && map[i][j-1] && !map[i-1][j-1])
+                        numOfIslands--;

```

Comment:
This could be rephrased as:
"if North and West and not NW" -> --
below can be rephrased as 
"if not N and  not W "  -> ++

I'm not sure these rules always work. For example, do they tell the difference between the following correctly

           XXXX       
X X X  X X X 
XXXX  XXXX 

Your code might work, but I can't tell it works by just reading the code and tests.